### PR TITLE
feat: mailerservice

### PR DESCRIPTION
**Pull request by felixleo22**

### CODE REVIEW

File: docker-compose.preprod.yml [hunk 1]
```diff
@@ -120,6 +117,27 @@ services:
     restart: unless-stopped
     networks:
       - elastic
+    
+  mail:
+    build:
+      context: ./src/auth
```
Comment (wilmouths) [inline docker-compose.preprod.yml:123]: I think it's the wrong context, maybe : ./src/mail ?

File: docker-compose.yml [hunk 2]
```diff
@@ -111,6 +108,27 @@ services:
       - redis
     restart: unless-stopped
 
+  mail:
+    build:
+      context: ./src/auth
```
Comment (wilmouths) [inline docker-compose.yml:113]: I think it's the wrong context, maybe : ./src/mail ?

File: rp/templates/dev.conf.template [hunk 3]
```diff
@@ -30,6 +30,10 @@ server {
     proxy_pass http://auth:7000/;
   }
 
+  location ^~ /api/mail/ {
+    proxy_pass http://mail:8000/;
+  }
+
```
Comment (wilmouths) [inline rp/templates/dev.conf.template:36]: Not useful if no external access needed

File: rp/templates/https.conf.template [hunk 4]
```diff
@@ -54,6 +54,10 @@ server {
     proxy_pass http://auth:7000/;
   }
 
+  location ^~ /api/mail/ {
+    proxy_pass http://mail:8000/;
+  }
+
```
Comment (wilmouths) [inline rp/templates/https.conf.template:60]: Not useful if no external access needed

File: src/mail/routers/mail.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  } catch (err) {
+    return next(err);
+  }
+  return res.status(202).json({});
+});
+
+// TODO
+// router.post('/report', async (req, res, next) => {});
+
+// router.post('/update-start', async (req, res, next) => {});
```
Comment (wilmouths) [inline src/mail/routers/mail.js:39]: To be removed if not used in the near future

File: src/mail/lib/logger.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+// TODO log for prod
+// function prodFormat() {
+//   const replaceError = ({
+//     name, level, message, stack,
+//   }) => ({
+//     name, level, message, stack,
+//   });
+//   const replacer = (key, value) => (value instanceof Error ? replaceError(value) : value);
+//   return combine(label({ name: 'ssr server log' }), format.json({ replacer }));
+// }
```
Comment (wilmouths) [inline src/mail/lib/logger.js:28]: To be removed if not used in the near future

File: src/mail/app.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+const outDir = path.resolve(__dirname, 'out');
+
+fs.ensureDir(path.resolve(outDir));
+fs.ensureDir(path.resolve(outDir, 'logs'));
+
+const app = express();
+
+app.use(cors({
+  origin: '*',
+  method: ['POST'],
```
Comment (wilmouths) [inline src/mail/app.js:21]: Add GET for / route

File: docker-compose.preprod.yml (hunk 1 above)
Comment (felixleo22) [reply docker-compose.preprod.yml:123]: i had fix it

**Pull request merged**